pkg/steps/output_image_tag.go

```diff
@@ -69,29 +55,49 @@ func (s *outputImageTagStep) Run(ctx context.Context, dry bool) error {
 		return nil
 	}
 
-	if err := s.istClient.ImageStreamTags(toNamespace).Delete(ist.Name, nil); err != nil && !errors.IsNotFound(err) {
-		return fmt.Errorf("could not remove output imagestreamtag: %v", err)
-	}
-	_, err := s.istClient.ImageStreamTags(toNamespace).Create(ist)
-	if err != nil && !errors.IsAlreadyExists(err) {
-		return fmt.Errorf("could not create output imagestreamtag: %v", err)
+	// Create if not exists, update if it does
+	if _, err := s.istClient.ImageStreamTags(toNamespace).Create(ist); err != nil {
+		if errors.IsAlreadyExists(err) {
+			existingIst, err := s.istClient.ImageStreamTags(ist.Namespace).Get(ist.Name, meta.GetOptions{})
+			if err != nil {
+				return fmt.Errorf("could not get existing output imagestreamtag for update: %v", err)
+			}
+			// We don't care about the existing imagestreamtag's state, we just
+			// want it to look like the new one, so we only copy the
+			// ResourceVersion so we can update it.
+			ist.ResourceVersion = existingIst.ResourceVersion
+			if _, err = s.istClient.ImageStreamTags(toNamespace).Update(ist); err != nil {
+				return fmt.Errorf("could not update output imagestreamtag: %v", err)
+			}
+		} else {
+			return fmt.Errorf("could not create output imagestreamtag: %v", err)
+		}
 	}
 	return nil
 }
 
 func (s *outputImageTagStep) Done() (bool, error) {
 	toNamespace := s.namespace()
 	log.Printf("Checking for existence of %s/%s:%s", toNamespace, s.config.To.Name, s.config.To.Tag)
-	if _, err := s.istClient.ImageStreamTags(toNamespace).Get(
-		fmt.Sprintf("%s:%s", s.config.To.Name, s.config.To.Tag),
-		meta.GetOptions{},
-	); err != nil {
+	ist, err := s.istClient.ImageStreamTags(toNamespace).Get(fmt.Sprintf("%s:%s", s.config.To.Name, s.config.To.Tag), meta.GetOptions{})
+	if err != nil {
 		if errors.IsNotFound(err) {
 			return false, nil
 		}
 		return false, fmt.Errorf("could not retrieve output imagestreamtag: %v", err)
 	}
-	return true, nil
+
+	// TODO(chance): this doesn't handle dry run since Done() doesn't have
+	// information about if it's a dry-run
+	from, err := s.istClient.ImageStreamTags(s.jobSpec.Namespace).Get(fmt.Sprintf("%s:%s", api.PipelineImageStream, s.config.From), meta.GetOptions{})
+	if err != nil {
+		return false, fmt.Errorf("could not resolve base image: %v", err)
+	}
+
+	desiredIst := s.imageStreamTag(from.Image.Name)
+	// if a tag already exists but doesn't match what we're looking for we're
+	// not done
+	return reflect.DeepEqual(ist.Tag, desiredIst.Tag), nil
```

Comment:
Maybe we should use semantic `DeepEqual` to compare api objects? 

https://godoc.org/k8s.io/apimachinery/pkg/api/equality

Reply:
Good idea